components/script/dom/node.rs

```diff
@@ -2625,6 +2777,22 @@ pub fn document_from_node<T: DerivedFrom<Node> + DomObject>(derived: &T) -> DomR
     derived.upcast().owner_doc()
 }
 
+pub fn shadow_root_from_node<T: DerivedFrom<Node> + DomObject>(
+    derived: &T,
+) -> Option<DomRoot<ShadowRoot>> {
+    derived.upcast().owner_shadow_root()
+}
+
+pub fn stylesheets_owner_from_node<T: DerivedFrom<Node> + DomObject>(
+    derived: &T,
+) -> Box<StyleSheetListOwner> {
```

Comment:
Out of curiosity, why a trait object with virtual dispatch instead of just an enum?